modules/cmf/src/main/java/org/jpos/cmf/AmountFee.java

```diff
@@ -0,0 +1,122 @@
+        feeType = FeeType.fromCode(data.substring(0, 2));
+        fee = new CMFAmount(data.substring(2, 15));
+        setConversionRate(data.substring(15, 23));
+        settlementFee = new CMFAmount(data.substring(23));
+    }
+
+    public FeeType getFeeType() {
+        return feeType;
+    }
+
+    public void setFeeType(FeeType feeType) {
+        this.feeType = requireNonNull(feeType);
+    }
+
+    public CMFAmount getFee() {
+        return fee;
+    }
+
+    public void setFee(CMFAmount fee) {
+        this.fee = requireNonNull(fee);
+    }
+
+    public CMFAmount getSettlementFee() {
+        return settlementFee;
+    }
+
+    public void setSettlementFee(CMFAmount settlementFee) {
+        this.settlementFee = requireNonNull(settlementFee);
+    }
+
+    public BigDecimal getConversionRate() {
+        return conversionRate;
+    }
+
+    public void setConversionRate(BigDecimal conversionRate) {
+        // BBB: what if the value has more than 7 significant digits (precision)?
+        //      should we round the value or reject it?
+        this.conversionRate = requireNonNull(conversionRate);
+    }
+
+    public void setConversionRate(String conversionRate) {
+        if (conversionRate != null) {
+            // Conversion rate is N8, where first digit moves the decimal point to the left
+            // But: ISO8583-1:2003 "6.2.4 Conversion rates"
+            //    If the value of the first digit is 8, a single zero after the decimal point is assumed.
+            //    If the value of the first digit is 9, two zeroes after the decimal point are assumed.
+            int conversionScale = conversionRate.charAt(0) - '0';
+            if (conversionScale == 8 || conversionScale == 9)
+                conversionScale++;
+            setConversionRate(new BigDecimal(new BigInteger(conversionRate.substring(1)), conversionScale));
+        }
+    }
+
+    public String serialize() {
+        // TODO: 1. Correctly serialize for cases with with high decimal counts
+        //       ----
+        //       2. Maybe we should check that the (unscaled) value has no more than 7 digits?
+        //         Should we round, fail?
+        return feeType.getCode() +
+                fee.serialize() +
+                conversionRate.scale() + ISOUtil.zeropad(conversionRate.unscaledValue().intValue(), 7) +
+                settlementFee.serialize();
+    }
+
+    // BBB comment on the RATE_PRECISION: currently == MathContext(7, RoundingMode.HALF_EVEN)
+    //     This is for the rare cases where the conversionRate has not been made explicit in the constructors.
+    //     Another alternative: Eliminate this method and DON'T ALLOW the construction of this object with a null conversion rate!
+    protected void calculateConversionRate() {
+        if (conversionRate == null)
+            conversionRate = settlementFee.getAmount().divide(fee.getAmount(), RATE_PRECISION);
+    }
```

Comment:
trust me, it's not a rare case if you are transforming from a legacy version, I would remove that method entirely and throw exception if conversion rat is null on `serialize`

Reply:
OK. Hago eso que me decís. 
Al borrar el método, desaparece el problema de la precisión y rounding.
Muerto el perro, se acabó la rabia :-)